code/espurna/serialsensor.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+
+SERIAL_SENSOR MODULE
+
+Copyright (C) 2017-2021 by Patrizio Bekerle <patrizio at bekerle dot com>
+
+*/
+
+#if SERIAL_SENSOR_SUPPORT
+
+#include "serialsensor.h"
+#include "mqtt.h"
+#include "ws.h"
+
+// -----------------------------------------------------------------------------
+// COMMUNICATIONS
+// -----------------------------------------------------------------------------
+
+/**
+ * Setup function
+ *
+ * I haven't managed to get the terminalRegisterCommand commands in a method
+ * because of the lambda function
+ */
+void serialSensorSetup() {
+#if TERMINAL_SUPPORT
+#if MQTT_SUPPORT
+    terminalRegisterCommand(F("SENSOR0"), [](const terminal::CommandContext& ctx) {
+        if (ctx.argc == 2) {
+            String value = String(ctx.argv[1]);
+            mqttSend("SENSOR0", value.c_str());
+            ctx.output.printf_P(PSTR("SENSOR0 has value %s\n"),
+                value.c_str());
+        }
+
+        terminalOK(ctx);
+    });
```

Comment:
b/c of the code repetition, maybe what we want is `mqtt.send` command with 2 args - 'topic' and 'payload'? could also just add it to the mqtt.cpp directly
also note of the 'retained' option, should these be ephemeral messages?

Reply:
> b/c of the code repetition, maybe what we want is mqtt.send command with 2 args - 'topic' and 'payload'? could also just add it to the mqtt.cpp directly

But how would I get the variables into the callback of `terminalRegisterCommand`? I couldn't sneak them in in the `[]`.

> also note of the 'retained' option, should these be ephemeral messages?

Yes, I guess they could be retained.

Reply:
> But how would I get the variables into the callback of terminalRegisterCommand? I couldn't sneak them in in the [].

Which ones though? I meant something like this

```cpp
terminalRegisterCommand(F("MQTT.SEND"), [](const terminal::CommandContext& ctx) {
    if (ctx.argc == 3) {
        if (mqttSend(ctx.argv[1].c_str(), ctx.argv[2].c_str(), false, false)) {
            terminalOK(ctx);
        } else {
            terminalError(ctx, F("Cannot queue the message")); //
        }
        return;
    }

    terminalError(ctx, F("MQTT.SEND <topic> <payload>"));
});
```

Unless we somehow want the command names to end up in the `commands` / `help`

Reply:
Yes, sounds like a great idea too. I'll try that later this week!